File: NetNgArch/src/ProjectBase/Web.Mvc/WebApplicationStorage.cs
using System;
using System.Collections.Generic;
using System.Web;
using System.Web.Configuration;
using ProjectBase.BusinessDelegate;

namespace ProjectBase.Web.Mvc
{


    public class WebApplicationStorage : IApplicationStorage
    {
        private readonly HttpApplicationState storage = null;

        public WebApplicationStorage(HttpApplicationState storage)
        {
            this.storage = storage;
        }
        public Object Get(string key)
        {
            return storage[key];
        }


        public void Set(string key, Object obj)
        {
            storage[key] = obj;
        }
        public String GetAppSetting(String key)
        {
            return WebConfigurationManager.AppSettings[key];
        }
        public String GetRealPath(String relativePath)
        {
            return HttpContext.Current.Server.MapPath(relativePath);
        }
    }
}

File: JavaNgArch/src/projectbase/mvc/WebApplicationStorage.java
package projectbase.mvc;

import javax.servlet.ServletContext;

import projectbase.bd.IApplicationStorage;

public class WebApplicationStorage implements IApplicationStorage{
	
    private ServletContext storage = null;

    public WebApplicationStorage(ServletContext storage)
    {
        this.storage = storage;
    }
    public Object Get(String key)
    {
        return storage.getAttribute(key);
    }


    public void Set(String key, Object obj)
    {
        storage.setAttribute(key, obj);
    }
    public String GetInitParameter(String key){
    	return storage.getInitParameter(key);
    }
    public String GetRealPath(String relativePath){
    	return storage.getRealPath(relativePath);
    }
}
